tests/unittests/test_exception_handling.py

```diff
@@ -0,0 +1,128 @@
+                       QuickbooksBadRequestError], 'HTTP-error-code: 400, Error: Message: Bad request, Quickbooks Error code: 400, Detail: Bad request for the URL'],
+        ['400_multiple_error', [400, {'Fault': {'Error': [{'Message': 'Bad request 1', 'code': '400', 'Detail': 'Bad request 1 for the URL'}, \
+            {'Message': 'Bad request 2', 'code': '400', 'Detail': 'Bad request 2 for the URL'}]}}, QuickbooksBadRequestError],
+            'HTTP-error-code: 400, Error: [{\'Message\': \'Bad request 1\', \'code\': \'400\', \'Detail\': \'Bad request 1 for the URL\'}, {\'Message\': \'Bad request 2\', \'code\': \'400\', \'Detail\': \'Bad request 2 for the URL\'}]'],
+        ['401_error', [401, {'fault': {'error': [{'Message': 'Not Authorized', 'code': '401', 'Detail': 'Not authorized to access'}]}},
+                       QuickbooksAuthenticationError], 'HTTP-error-code: 401, Error: Message: Not Authorized, Quickbooks Error code: 401, Detail: Not authorized to access'],
+        ['403_error', [403, {'Fault': {'Error': [{'Message': 'Forbidden', 'code': '403', 'Detail': 'Forbidden for the URL'}]}},
+                       QuickbooksForbiddenError], 'HTTP-error-code: 403, Error: Message: Forbidden, Quickbooks Error code: 403, Detail: Forbidden for the URL'],
+        ['404_error', [404, {'Fault': {'Error': [{'Message': 'Not found', 'code': '404', 'Detail': 'Not found for the URL'}]}},
+                       QuickbooksNotFoundError], 'HTTP-error-code: 404, Error: Message: Not found, Quickbooks Error code: 404, Detail: Not found for the URL'],
+        ['500_error', [500, {'Fault': {'Error': [{'Message': 'Internal Server error', 'code': '500', 'Detail': 'The service is unreachable'}]}},
+                       QuickbooksInternalServerError], 'HTTP-error-code: 500, Error: Message: Internal Server error, Quickbooks Error code: 500, Detail: The service is unreachable'],
+        ['501_error', [501, {'Fault': {'Error': [{'Message': 'Not implemented', 'code': '501', 'Detail': 'Not implemented for the URL'}]}},
+                       Quickbooks5XXException], 'HTTP-error-code: 501, Error: Message: Not implemented, Quickbooks Error code: 501, Detail: Not implemented for the URL'],
+        ['503_error', [503, {'Fault': {'Error': [{'Message': 'Service unavailable', 'code': '503', 'Detail': 'The service is not available'}]}},
+                       QuickbooksServiceUnavailableError], 'HTTP-error-code: 503, Error: Message: Service unavailable, Quickbooks Error code: 503, Detail: The service is not available'],
+    ])
+    @mock.patch('time.sleep')
+    @mock.patch('tap_quickbooks.client.QuickbooksClient._write_config')
+    @mock.patch('tap_quickbooks.client.QuickbooksClient.get')
+    @mock.patch('requests_oauthlib.OAuth2Session.request')
+    def test_API_exception_handling(self, test_name, test_data, exp, mocked_oauth_request, mocked_get, mocked_write_config, mock_sleep):
```

Comment:
Is this the correct order for the decorators? If so, is there a way to make this more clear? test_API_exception_handling has arguments in an order that doesn't quite match the above given that the parameterized expand would be test_name, test_data, exp, but then decorators for mocked_oauth_request through sleep are in reversed order from the function args